moveit_ros/planning/planning_scene_monitor/src/current_state_monitor.cpp

```diff
@@ -92,24 +92,16 @@ ros::Time CurrentStateMonitor::getCurrentStateTimeHelper(const std::string& grou
       return ros::Time(0.0);
     }
   }
-  auto oldest_state_time = ros::Time();
+  auto oldest_state_time = ros::Time::now();
   for (const moveit::core::JointModel* joint : *active_joints)
   {
     auto it = joint_time_.find(joint);
     if (it == joint_time_.end())
     {
       ROS_DEBUG_NAMED(LOGNAME, "Joint '%s' has never been updated", joint->getName().c_str());
     }
-    else
-    {
-      if (!oldest_state_time.isZero())
```

Comment:
I'm afraid you cannot simplify away this code as some tests (and existing code) rely on a return value of zero when no joints were published yet. Otherwise, I agree to your point.

Reply:
I see the problem, but I dislike the current logic as it fails in the trivial case (no active joint in the group or only static transforms in the group).

I am confused why `Joint '%s' has never been updated` is only a debug output here instead of a reason to return 0 then. @tahsinkose @rhaschke maybe you can comment why this condition was handled implicitly in #3580 and whether it's feasible for your use-cases to change this logic instead.

Reply:
I think the _helper_ should not generate any visible log message. Missing joints are handled upstream in the call hierarchy. As your initial comment was not very clear, I'll try to summarize our goals here:
1. getCurrentStateTimeHelper should return the oldest time a joint from the group was updated
2. If there are joints that were never updated, the return value should be zero
3. Joint updates received via `tf_static` should be always considered recent
  This is the key feature you were missing. Those joints, having zero `joint_time_` resulted in the function incorrectly returning zero, right?

Apart from `tf_static` joints, which were probably ignored when designing the code, the current code (and the tests) also violate the second condition. So far, the oldest time of all _received_ joints (in the group) was returned. However, this causes `waitForCurrentState()` to ignore missing joints:
https://github.com/moveit/moveit/blob/675495426dcceb18453ab0a689f5fd8319248998/moveit_ros/planning/planning_scene_monitor/src/current_state_monitor.cpp#L278

I pushed a new proposal.

Reply:
friendly ping @v4hn

Reply:
Indeed. I didn't have `tf_static` in mind at the time I pushed https://github.com/moveit/moveit/pull/3580. The rationale behind the log `Joint '%s' has never been updated` being only a debug output seems to provide a log-level overall sanity check of the joint states machinery. A user could see all problematic joints in one robot session this way, whereas returning immediately at L102 would make that a bit more uncomfortable -- i.e. one at a time.